src/appointment/appointment.service.ts

```
@@ -0,0 +1,116 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common';
+import { InjectModel } from 'nestjs-typegoose';
+import { Appointment } from 'src/model/appointment.model';
+import { AppointmentStatus } from 'src/enum/appointment.enum';
+import { ReturnModelType } from '@typegoose/typegoose';
+import { CreateAppointmentDTO, EditAppointmentDTO } from './appointment.dto';
+import { UserService } from 'src/user/user.service';
+import { UserRole } from 'src/enum/user.enum';
+
+@Injectable()
+export class AppointmentService {
+    constructor(
+        @InjectModel(Appointment)
+        private readonly model: ReturnModelType<typeof Appointment>,
+        private readonly userService: UserService,
+    ) {}
+
+    async createAppointment(
+        { tutorId, ...createAppointmentDTO }: CreateAppointmentDTO,
+        studentId: string,
+    ): Promise<Appointment> {
+        const tutor = await this.userService.findById(tutorId);
+        if (tutor.role !== UserRole.Tutor)
+            throw new HttpException('Invalid tutorId', HttpStatus.NOT_FOUND);
```

Should throw `NotFoundException('Invalid tutorId')`